### ISSUE TITLE
sepolia node stalled at block 1460320

### ISSUE THREAD

**Issue by mcdee** [hyperledger/besu#4076]
### Description

I have two besu/lighthouse nodes which have stalled at block 1460320.  A besu/prysm node has been reported has having the same issue (see https://forkmon.sepolia.ethdevops.io/ for details).

### Logs
```
2022-07-08 01:28:48.669+00:00 | vert.x-worker-thread-0 | INFO  | EngineNewPayload | New payload: number: 1460317, hash: 0x01c4b2cae4e9dd7865a7134c67c2cdff8ebb842c0d92fb4b5757ede12fad8894, parentHash: 0x09ca77408a4a3c3639b5d0ba281d75bb2cd9dc6138ef4d16034d3fbd61d164b9, latestValidHash: 0x01c4b2cae4e9dd7865a7134c67c2cdff8ebb842c0d92fb4b5757ede12fad8894, status: VALID
2022-07-08 01:28:48.674+00:00 | vert.x-worker-thread-0 | INFO  | EngineForkchoiceUpdated | Consensus fork-choice-update: head: 0x01c4b2cae4e9dd7865a7134c67c2cdff8ebb842c0d92fb4b5757ede12fad8894, finalized: 0x0d57abf35768ee84c91c7f9756613bb46913b723c55e98862752f23d25ad6aeb, safeBlockHash: 0x01c4b2cae4e9dd7865a7134c67c2cdff8ebb842c0d92fb4b5757ede12fad8894
2022-07-08 01:29:00.140+00:00 | vert.x-worker-thread-0 | INFO  | EngineNewPayload | New payload: number: 1460318, hash: 0xe3e09fc16ba7423baf3688ddfe04af336e63eaacc4edabfbbe60188f8cd6043c, parentHash: 0x01c4b2cae4e9dd7865a7134c67c2cdff8ebb842c0d92fb4b5757ede12fad8894, latestValidHash: 0xe3e09fc16ba7423baf3688ddfe04af336e63eaacc4edabfbbe60188f8cd6043c, status: VALID
2022-07-08 01:29:00.144+00:00 | vert.x-worker-thread-0 | INFO  | EngineForkchoiceUpdated | Consensus fork-choice-update: head: 0xe3e09fc16ba7423baf3688ddfe04af336e63eaacc4edabfbbe60188f8cd6043c, finalized: 0x0d57abf35768ee84c91c7f9756613bb46913b723c55e98862752f23d25ad6aeb, safeBlockHash: 0xe3e09fc16ba7423baf3688ddfe04af336e63eaacc4edabfbbe60188f8cd6043c
2022-07-08 01:29:10.901+00:00 | Timer-0 | INFO  | DNSTimerTask | Refreshing DNS records with enrtree://AKA3AM6LPBYEUDMVNU3BSVQJ5AD45Y7YPOHJLEF6W26QOE4VTUDPE@all.sepolia.ethdisco.net
2022-07-08 01:29:13.494+00:00 | vert.x-worker-thread-0 | INFO  | EngineNewPayload | New payload: number: 1460318, hash: 0x7c03511ab247bb603297d6d3ca225757ca9fde5d29f0f15e888024e82d3502c2, parentHash: 0x01c4b2cae4e9dd7865a7134c67c2cdff8ebb842c0d92fb4b5757ede12fad8894, latestValidHash: 0x7c03511ab247bb603297d6d3ca225757ca9fde5d29f0f15e888024e82d3502c2, status: VALID
2022-07-08 01:29:24.080+00:00 | vert.x-worker-thread-0 | INFO  | EngineNewPayload | New payload: number: 1460319, hash: 0xa605427a60efdd08a1853e03162af689a19a57909799e03031b104f833d8c145, parentHash: 0xe3e09fc16ba7423baf3688ddfe04af336e63eaacc4edabfbbe60188f8cd6043c, latestValidHash: 0xa605427a60efdd08a1853e03162af689a19a57909799e03031b104f833d8c145, status: VALID
2022-07-08 01:29:24.084+00:00 | vert.x-worker-thread-0 | INFO  | EngineForkchoiceUpdated | Consensus fork-choice-update: head: 0xa605427a60efdd08a1853e03162af689a19a57909799e03031b104f833d8c145, finalized: 0x0d57abf35768ee84c91c7f9756613bb46913b723c55e98862752f23d25ad6aeb, safeBlockHash: 0xa605427a60efdd08a1853e03162af689a19a57909799e03031b104f833d8c145
2022-07-08 01:29:36.156+00:00 | vert.x-worker-thread-0 | INFO  | EngineNewPayload | New payload: number: 1460320, hash: 0x0907ccc11e8c811f6e5271e870f3c5a81c72c5c3ffd1485544babfb833f5365c, parentHash: 0xa605427a60efdd08a1853e03162af689a19a57909799e03031b104f833d8c145, latestValidHash: 0x0907ccc11e8c811f6e5271e870f3c5a81c72c5c3ffd1485544babfb833f5365c, status: VALID
2022-07-08 01:29:36.164+00:00 | vert.x-worker-thread-0 | INFO  | EngineForkchoiceUpdated | Consensus fork-choice-update: head: 0x0907ccc11e8c811f6e5271e870f3c5a81c72c5c3ffd1485544babfb833f5365c, finalized: 0x0d57abf35768ee84c91c7f9756613bb46913b723c55e98862752f23d25ad6aeb, safeBlockHash: 0x0907ccc11e8c811f6e5271e870f3c5a81c72c5c3ffd1485544babfb833f5365c
2022-07-08 01:29:49.515+00:00 | vert.x-worker-thread-0 | INFO  | EngineNewPayload | New payload: number: 1460319, hash: 0xba7875dcdc7d497fdaea4cf7495802cae87d78288b20734e8f3327cb66e77d1d, parentHash: 0x7c03511ab247bb603297d6d3ca225757ca9fde5d29f0f15e888024e82d3502c2, latestValidHash: 0xba7875dcdc7d497fdaea4cf7495802cae87d78288b20734e8f3327cb66e77d1d, status: VALID
java.lang.RuntimeException: java.lang.IllegalStateException: Address=0xcd8080b5a4e828232a6390fb0d5dca790ae90c85 Prior Value in Rolling Change: nonces differ
        at org.hyperledger.besu.ethereum.bonsai.BonsaiWorldStateArchive.getMutable(BonsaiWorldStateArchive.java:254)
        at org.hyperledger.besu.ethereum.MainnetBlockValidator.validateAndProcessBlock(MainnetBlockValidator.java:100)
        at org.hyperledger.besu.consensus.merge.blockcreation.MergeCoordinator.validateBlock(MergeCoordinator.java:256)
        at org.hyperledger.besu.consensus.merge.blockcreation.MergeCoordinator.rememberBlock(MergeCoordinator.java:277)
        at org.hyperledger.besu.consensus.merge.blockcreation.TransitionCoordinator.rememberBlock(TransitionCoordinator.java:137)
        at org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.engine.EngineNewPayload.syncResponse(EngineNewPayload.java:185)
        at org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.ExecutionEngineJsonRpcMethod.lambda$response$0(ExecutionEngineJsonRpcMethod.java:62)
        at io.vertx.core.impl.ContextImpl.lambda$null$0(ContextImpl.java:159)
        at io.vertx.core.impl.AbstractContext.dispatch(AbstractContext.java:100)
        at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$1(ContextImpl.java:157)
        at io.vertx.core.impl.TaskQueue.run(TaskQueue.java:76)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.IllegalStateException: Address=0xcd8080b5a4e828232a6390fb0d5dca790ae90c85 Prior Value in Rolling Change: nonces differ
        at org.hyperledger.besu.ethereum.bonsai.BonsaiAccount.assertCloseEnoughForDiffing(BonsaiAccount.java:303)
        at org.hyperledger.besu.ethereum.bonsai.BonsaiWorldStateUpdater.rollAccountChange(BonsaiWorldStateUpdater.java:497)
        at org.hyperledger.besu.ethereum.bonsai.BonsaiWorldStateUpdater.lambda$rollForward$9(BonsaiWorldStateUpdater.java:422)
        at java.base/java.util.HashMap$EntrySpliterator.forEachRemaining(HashMap.java:1850)
        at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:762)
        at org.hyperledger.besu.ethereum.bonsai.BonsaiWorldStateUpdater.rollForward(BonsaiWorldStateUpdater.java:420)
        at org.hyperledger.besu.ethereum.bonsai.BonsaiWorldStateArchive.getMutable(BonsaiWorldStateArchive.java:243)
        ... 14 more
2022-07-08 01:30:00.088+00:00 | vert.x-worker-thread-0 | ERROR | MainnetBlockValidator | Unable to process block because parent world state 0x0f4b525b190ce9de7c11c57e0fc2d1e1fa2133d753b02686a4e3495135cf1703 is not available. Block 1460321 (0xab50faca6cde7f7703badd3f51ab1d5a42d0358d6a30fde2a5d8908089e1e95e)
2022-07-08 01:30:00.089+00:00 | vert.x-worker-thread-0 | INFO  | EngineNewPayload | Invalid new payload: number: 1460321, hash: 0xab50faca6cde7f7703badd3f51ab1d5a42d0358d6a30fde2a5d8908089e1e95e, parentHash: 0x0907ccc11e8c811f6e5271e870f3c5a81c72c5c3ffd1485544babfb833f5365c, latestValidHash: 0x0907ccc11e8c811f6e5271e870f3c5a81c72c5c3ffd1485544babfb833f5365c, status: INVALID, validationError: Unable to process block because parent world state 0x0f4b525b190ce9de7c11c57e0fc2d1e1fa2133d753b02686a4e3495135cf1703 is not available
java.lang.RuntimeException: java.lang.IllegalStateException: Address=0xcd8080b5a4e828232a6390fb0d5dca790ae90c85 Prior Value in Rolling Change: nonces differ
        at org.hyperledger.besu.ethereum.bonsai.BonsaiWorldStateArchive.getMutable(BonsaiWorldStateArchive.java:254)
        at org.hyperledger.besu.ethereum.MainnetBlockValidator.validateAndProcessBlock(MainnetBlockValidator.java:100)
        at org.hyperledger.besu.consensus.merge.blockcreation.MergeCoordinator.validateBlock(MergeCoordinator.java:256)
        at org.hyperledger.besu.consensus.merge.blockcreation.MergeCoordinator.rememberBlock(MergeCoordinator.java:277)
        at org.hyperledger.besu.consensus.merge.blockcreation.TransitionCoordinator.rememberBlock(TransitionCoordinator.java:137)
        at org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.engine.EngineNewPayload.syncResponse(EngineNewPayload.java:185)
        at org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.ExecutionEngineJsonRpcMethod.lambda$response$0(ExecutionEngineJsonRpcMethod.java:62)
        at io.vertx.core.impl.ContextImpl.lambda$null$0(ContextImpl.java:159)
        at io.vertx.core.impl.AbstractContext.dispatch(AbstractContext.java:100)
        at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$1(ContextImpl.java:157)
        at io.vertx.core.impl.TaskQueue.run(TaskQueue.java:76)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.IllegalStateException: Address=0xcd8080b5a4e828232a6390fb0d5dca790ae90c85 Prior Value in Rolling Change: nonces differ
        at org.hyperledger.besu.ethereum.bonsai.BonsaiAccount.assertCloseEnoughForDiffing(BonsaiAccount.java:303)
        at org.hyperledger.besu.ethereum.bonsai.BonsaiWorldStateUpdater.rollAccountChange(BonsaiWorldStateUpdater.java:497)
        at org.hyperledger.besu.ethereum.bonsai.BonsaiWorldStateUpdater.lambda$rollForward$9(BonsaiWorldStateUpdater.java:422)
        at java.base/java.util.HashMap$EntrySpliterator.forEachRemaining(HashMap.java:1850)
        at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:762)
        at org.hyperledger.besu.ethereum.bonsai.BonsaiWorldStateUpdater.rollForward(BonsaiWorldStateUpdater.java:420)
        at org.hyperledger.besu.ethereum.bonsai.BonsaiWorldStateArchive.getMutable(BonsaiWorldStateArchive.java:243)
        ... 14 more
2022-07-08 01:30:00.108+00:00 | vert.x-worker-thread-0 | ERROR | MainnetBlockValidator | Unable to process block because parent world state 0x0f4b525b190ce9de7c11c57e0fc2d1e1fa2133d753b02686a4e3495135cf1703 is not available. Block 1460321 (0xab50faca6cde7f7703badd3f51ab1d5a42d0358d6a30fde2a5d8908089e1e95e)
2022-07-08 01:30:00.108+00:00 | vert.x-worker-thread-0 | INFO  | EngineNewPayload | Invalid new payload: number: 1460321, hash: 0xab50faca6cde7f7703badd3f51ab1d5a42d0358d6a30fde2a5d8908089e1e95e, parentHash: 0x0907ccc11e8c811f6e5271e870f3c5a81c72c5c3ffd1485544babfb833f5365c, latestValidHash: 0x0907ccc11e8c811f6e5271e870f3c5a81c72c5c3ffd1485544babfb833f5365c, status: INVALID, validationError: Unable to process block because parent world state 0x0f4b525b190ce9de7c11c57e0fc2d1e1fa2133d753b02686a4e3495135cf1703 is not available
java.lang.RuntimeException: java.lang.IllegalStateException: Address=0xcd8080b5a4e828232a6390fb0d5dca790ae90c85 Prior Value in Rolling Change: nonces differ
...
```

### Versions (Add all that apply)
* Software version: besu/v22.7.0-RC1/linux-x86_64/openjdk-java-17

**Comment by daniellehrner** [comment]
I think this was just fixed yesterday: [#4069](https://github.com/hyperledger/besu/pull/4069). We are in the process of the final tests of a hotfix release (22.4.4) which will released by the end of the day.

Unfortunately once this happens, the database is corrupted and we can not recover from that. The only way forward is to sync the node from scratch again.

**Comment by mcdee** [comment]
Does this really require a complete resync?  For this and similar problems, is there no way in which the database can be pruned such that it is rolled back to a known good (user-provided) height?

**Comment by mcdee** [comment]
Will there also be a release for the 22.7.0-RC1, given that this fixes other known issues with bonsai?

**Comment by garyschulte** [comment]
We discussed a hotfix to the 22.7.0-RC series to include #4069 but decided to instead aggregate all of the bonsai and critical merge fixes in a final 22.4.4 release.  The next RC series release will be July 20th.  

Most of the merge related difference between what is in 22.7 series that is not in 22.4.4 are fixes for corner cases and hive suite compliance.  @mcdee If there is a strong desire to run the 22.7 release candidate series, I think we could expedite an offcycle RC2

**Comment by garyschulte** [comment]
Closing 'prior value rolling change' as fixed with #4069.
